### DISCUSSION TITLE
CurrentDomainException

### DISCUSSION THREAD

**Discussion by Kerry-Hinge** [terjeio/ioSender#349] [Q&A] [answered]
As soon as IOSender starts I get the following dialog box appear:
CurrentDomainException
Unhandled exception occured: Input string was not in a correct format.
Clicking "OK" closes the dialog box and shut down the program. If I don't click "OK" and click on the program, sending the dialog box to the background, I can select and navigate within the program.

I have just built a new computer, an Intel NUC i3 system running Windows 10. The computer has nothing else on it and this is the first time I have used IOSender. Previously I have used Candle on a Windows 7 computer connected to a 3018 Mini CNC machine using an Arduino Nano. The connection is via a USB cable and during the initial start of IOSender, I selected the detected USB device but I still keep getting this error. Can someone explain what this error means and how to resolve it?

**Comment by terjeio** [comment]
Is the controller running Grbl?
If so which version? Post the `$I` output when connecting with Candle or from a terminal program such as putty.

**Reply by Kerry-Hinge** [reply]
I see that could be the problem. I have the 3018 CNC setup to run with an old VIAO laptop running Windows XP. It took me quite a bit of work to get all the versions of software just right for it to run but now I reconnect it to my Windows 7 computer, it no longer works correctly with Candle. I reconnected to the Windows XP computer and it is communicating correctly. The version of GRBL running on the Nano is $I < [0.9j.20160726:]. I am assuming I will need to upload a later version of GRBL for it to work with the Windows 10 computer.

**Reply by terjeio** [reply, marked as answer]
Grbl 0.9 is not supported by ioSender, you will have to update to 1.1. I guess I should add a check for the version instead of having the sender crashing...

**Comment by phil-barrett** [comment]
Grbl 0.9 is positively ancient. Could be why Candle is also failing. Be aware that updating to 1.1 will change the pin out - Z limit and spindle are swapped. At least they are on the Uno.  I believe the Nano version uses the same pins but you should verify that.

**Reply by Kerry-Hinge** [reply]
Thanks Phil but I have to use GRBL 0.9 to run with my old laptop on Windows XP with GRBL Control 0.8.1. It was such a pain setting up when I broke the USB connector on the Nano and bought a new Nano. I had to use an Uno connected to the Nano to flash an old version bootloader before I could even flash GRBL 0.9. Now that I have reconnected the 3018 CNC to my other computer running Windows 7 and Candle 1.1.7, I am faced with having to go through the whole process again of flashing the bootloader so instead I am just buying another Nano that I can swap in and out depending on which computer I am using with the CNC.
